backend/views/views.py

```diff
@@ -34,12 +36,43 @@ def admin():
     return render_template('admin_page.html')
 
 
-@app.route('/userpage')
+@app.route('/userpage', methods=['GET'])
 @admin_permissions
 def user_page():
+    form = SearchForm(request.form)
     users = db.session.query(User, Role).filter(
         User.role_id == Role.id).order_by(User.id).all()
-    return render_template('user_page.html', users=users)
+    if request.args.get('search') != "" and request.args.get('search') >= MIN_SEARCH_STR:
+    # if form.validate_on_submit():
+        key = int(request.args.get('field_by'))
+        search_string = str(request.args.get('search'))
+        search_paremeters = []
+        search_users = []
+        for one_string in search_string.split(' '):
+            if len(one_string) >= MIN_SEARCH_STR:
+                search_paremeters.append(''.join(["%", one_string, "%"]))
+        for search_paremeter in search_paremeters:
+            conditions = {1:"users.name LIKE  '%s'" %(search_paremeter),
```

Comment:
It's not good idea to use strings to form DB query as it may cause SQL-Injection vulnerability. Please use SQLAlchemy's built in [functions](http://docs.sqlalchemy.org/en/latest/orm/tutorial.html#common-filter-operators) for this.

For example this line may be re-written as:
```
{1: User.name.like(search_parameter),
```
`OR` statements below as `or_(conditions[1], conditions[2])`